### PR TITLE
[Dynamic Margin Calculation] Base configuration

### PR DESCRIPTION
**Pull request by thangqp**

### CODE REVIEW

File: src/test/java/org/gridsuite/dynamicmargincalculation/server/service/DynamicMarginCalculationResultServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+@SpringBootTest
+class DynamicMarginCalculationResultServiceTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DynamicMarginCalculationResultServiceTest.class);
+
+    @Autowired
+    DynamicMarginCalculationResultRepository resultRepository;
+
+    @Autowired
+    DynamicMarginCalculationResultService dynamicSecurityAnalysisResultService;
```
Comment (etiennehomer) [inline src/test/java/org/gridsuite/dynamicmargincalculation/server/service/DynamicMarginCalculationResultServiceTest.java]:
```suggestion
    DynamicMarginCalculationResultService dynamicMarginCalculationResultService;
```

File: src/main/java/org/gridsuite/dynamicmargincalculation/server/RestResponseEntityExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * @author Thang PHAM <quyet-thang.pham at rte-france.com>
+ */
+@ControllerAdvice
+public class RestResponseEntityExceptionHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RestResponseEntityExceptionHandler.class);
+
+    @ExceptionHandler(DynamicMarginCalculationException.class)
+    protected ResponseEntity<Object> handleDynamicSecurityAnalysisException(DynamicMarginCalculationException exception) {
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/dynamicmargincalculation/server/RestResponseEntityExceptionHandler.java]:
```suggestion
    protected ResponseEntity<Object> handleDynamicMarginCalculationException(DynamicMarginCalculationException exception) {
```

File: src/test/java/org/gridsuite/dynamicmargincalculation/server/service/DynamicMarginCalculationResultServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    @AfterEach
+    void cleanDB() {
+        resultRepository.deleteAll();
+    }
+
+    @Test
+    void testCrud() {
```
Comment (etiennehomer) [inline src/test/java/org/gridsuite/dynamicmargincalculation/server/service/DynamicMarginCalculationResultServiceTest.java]: It's heavy to add all those tests for an "empty" server. You could make it simpler before you really code into your DMC server

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/test/java/org/gridsuite/dynamicmargincalculation/server/service/DynamicMarginCalculationResultServiceTest.java (hunk 3 above)
Comment (thangqp) [reply src/test/java/org/gridsuite/dynamicmargincalculation/server/service/DynamicMarginCalculationResultServiceTest.java]: In my opinion, the result service is useful for verifying whether Spring Framework configurations (such as JPA and database settings) are functioning correctly. 

Additionally, it helps reduce the amount of code in the next ticket, which will focus more on functional aspects.

File: src/test/java/org/gridsuite/dynamicmargincalculation/server/service/DynamicMarginCalculationResultServiceTest.java (hunk 1 above)
Comment (thangqp) [reply src/test/java/org/gridsuite/dynamicmargincalculation/server/service/DynamicMarginCalculationResultServiceTest.java]: DONE in https://github.com/gridsuite/dynamic-margin-calculation-server/pull/1/commits/5c376744edd9fa5afe1e568d675b18068e80f938

File: src/main/java/org/gridsuite/dynamicmargincalculation/server/RestResponseEntityExceptionHandler.java (hunk 2 above)
Comment (thangqp) [reply src/main/java/org/gridsuite/dynamicmargincalculation/server/RestResponseEntityExceptionHandler.java]: DONE in https://github.com/gridsuite/dynamic-margin-calculation-server/pull/1/commits/5c376744edd9fa5afe1e568d675b18068e80f938

**Review by etiennehomer — APPROVED** [review]

File: src/test/java/org/gridsuite/dynamicmargincalculation/server/service/DynamicMarginCalculationResultServiceTest.java (hunk 3 above)
Comment (thangqp) [reply src/test/java/org/gridsuite/dynamicmargincalculation/server/service/DynamicMarginCalculationResultServiceTest.java]: MarginCalculationResult contains a list of LoadIncreaseResult in which we have a Dynawo status, see https://github.com/powsybl/powsybl-dynawo/blob/main/dynawo-algorithms/dynawo-margin-calculation/src/main/java/com/powsybl/dynawo/margincalculation/results/MarginCalculationResult.java. 

For this reason, it is better to separate GridSuite status into a separated table (as we done in LoadFlow), and all related to margin calculation result details in another table in the future

I have updated my PR with the commit here https://github.com/gridsuite/dynamic-margin-calculation-server/pull/1/commits/75e610f0dc7b71a9345f0779ff23c228c63738fc

**Review by etiennehomer — APPROVED** [review]
